sopht_mpi/numeric/eulerian_grid_ops/poisson_solver_2d/UnboundedPoissonSolverMPI2D.py

```diff
@@ -0,0 +1,346 @@
+            gen_elementwise_complex_product_pyst_kernel_2d(real_t=self.real_t)
+        )
+
+    def solve(self, solution_field, rhs_field):
+        """Unbounded Poisson solver method.
+        Solves Poisson equation in 2D: -del^2(solution_field) = rhs_field
+        for unbounded domain using Greens function convolution and
+        domain doubling trick (Hockney and Eastwood).
+        """
+
+        self.set_fixed_val_kernel_2d(field=self.domain_doubled_buffer, fixed_val=0)
+        self.mpi_domain_doubling_comm.copy_to_doubled_domain(
+            local_field=rhs_field,
+            local_doubled_field=self.domain_doubled_buffer,
+        )
+
+        self.rfft(
+            field=self.domain_doubled_buffer,
+            fourier_field=self.domain_doubled_fourier_buffer,
+        )
+
+        # Greens function convolution
+        self.elementwise_complex_product_kernel_2d(
+            product_field=self.convolution_buffer,
+            field_1=self.domain_doubled_fourier_buffer,
+            field_2=self.fourier_greens_function_times_dx_squared,
+        )
+
+        self.irfft(
+            fourier_field=self.convolution_buffer,
+            inv_fourier_field=self.domain_doubled_buffer,
+        )
+
+        self.mpi_domain_doubling_comm.copy_from_doubled_domain(
+            local_doubled_field=self.domain_doubled_buffer,
+            local_field=solution_field,
+        )
+
+
+class MPIDomainDoublingCommunicator2D:
+    """
+    Class exclusive for field communication between actual and doubled domain.
+    Since mpi4py-fft always require that at least one axis of a multidimensional
+    array remains aligned (non-distributed), in 2D that translates to slab
+    decomposition, which will be assumed for the communication operations here.
+    """
+
+    def __init__(self, ghost_size, mpi_construct):
+        self.mpi_construct = mpi_construct
+        # Use ghost_size to define offset indices for inner cell
+        if ghost_size < 0 and not isinstance(ghost_size, int):
+            raise ValueError(
+                f"Ghost size {ghost_size} needs to be an integer >= 0"
+                "for field communication."
+            )
+        self.ghost_size = ghost_size
+
+        # define local grid sizes for actual and doubled domain
+        self.field_grid_size = mpi_construct.local_grid_size
+        self.field_doubled_grid_size = mpi_construct.local_grid_size * 2
+
+        distributed_dim = np.where(np.array(mpi_construct.grid_topology) != 1)[0]
+        if len(distributed_dim) > 1:
+            raise ValueError(
+                f"Distributed in {len(distributed_dim)} dimensions."
+                "Only 1 dimension is allowed to be distributed in 2D (slab decomp)"
+            )
+        self.distributed_dim = distributed_dim[0]
+        # Copying from actual to doubled domain -> two receives, one send
+        # Copying from doubled to actual domain -> two sends, one receive
+        # Total requests needed = 3 requests
```

Comment:
why this is the case is not clear, maybe a brief discussion on this and then modifying comments will help.

Reply:
Added in PR description an attachment of a rough cartoon on the communication pattern.